# HG changeset patch
# User kai@linux-pc
# Date 1204148183 -3600
#      Wed Feb 27 22:36:23 2008 +0100
# Node ID 6acd0a18a3ee08cdc41b57cecd7e73c4ec4d33b4
# Parent  3422f39573b18cbab5d4731a92e7fad3f53d61de
If FaceColor is none don't use pm3d mode and set linestyle correctly.

diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -918,19 +918,58 @@ function __go_draw_axes__ (h, plot_strea
 					   && all (obj.facecolor == 1)));
 	    if (strncmp (obj.facecolor, "none", 4))
 	      hidden_removal = false;
 	    else
 	      if (isnan (hidden_removal))
 		hidden_removal = true;
 	      endif
 	    endif
-	    
+	
             if (flat_interp_edge && facecolor_none_or_white)
 	      withclause{data_idx} = "with line palette";
+	      fputs (plot_stream, "unset pm3d\n");
+	    elseif (facecolor_none_or_white)
+	      edgecol = obj.edgecolor;
+	      if (have_newer_gnuplot)
+		if (mono)
+		  colorspec = "";
+		else
+		  colorspec = sprintf ("linecolor rgb \"#%02x%02x%02x\"",
+				 round (255*edgecol));
+		endif
+                fprintf (plot_stream,
+                         "set style line %d %s lw %f;\n",
+                         data_idx, colorspec, obj.linewidth);
+              else
+		if (isequal (edgecol, [0,0,0]))
+		  typ = -1;
+		elseif (isequal (edgecol, [1,0,0]))
+		  typ = 1;
+		elseif (isequal (edgecol, [0,1,0]))
+		  typ = 2;
+		elseif (isequal (edgecol, [0,0,1]))
+		  typ = 3;
+		elseif (isequal (edgecol, [1,0,1]))
+		  typ = 4;
+		elseif (isequal (edgecol, [0,1,1]))
+		  typ = 5;
+		elseif (isequal (edgecol, [1,1,1]))
+		  typ = -1;
+		elseif (isequal (edgecol, [1,1,0]))
+		  typ = 7;
+		else
+		  typ = -1;
+		endif
+                fprintf (plot_stream,
+                         "set style line %d lt %d lw %f;\n",
+                         data_idx, typ, obj.linewidth);
+	      endif
+	      withclause{data_idx} = sprintf("with line linestyle %d", data_idx);
+	      fputs (plot_stream, "unset pm3d\n")
             endif
 
 	    if (have_newer_gnuplot)
 	      dord = "depthorder";
 	    else
 	      dord = "";
 	    endif
 
